src/main.rs

```diff
@@ -25,46 +25,55 @@ fn agent_start(settings_data: Settings) -> Result<Vec<JoinHandle<()>>, Error> {
     let token = settings_data.openbas.token;
     let unsecured_certificate = settings_data.openbas.unsecured_certificate;
     let with_proxy = settings_data.openbas.with_proxy;
-    let keep_alive_thread = keep_alive::ping(url.clone(), token.clone(), unsecured_certificate.clone(), with_proxy.clone());
+    let keep_alive_thread = keep_alive::ping(
+        url.clone(),
+        token.clone(),
+        unsecured_certificate,
+        with_proxy,
+    );
     // Starts the agent listening thread
-    let agent_job_thread = agent_job::listen(url.clone(), token.clone(), unsecured_certificate.clone(), with_proxy.clone());
+    let agent_job_thread = agent_job::listen(
+        url.clone(),
+        token.clone(),
+        unsecured_certificate,
+        with_proxy,
+    );
     // Starts the cleanup thread
     let cleanup_thread = agent_cleanup::clean();
     // Don't stop the exec until the listening thread is done
-    return Ok(vec![
-        keep_alive_thread.unwrap(), 
-        agent_job_thread.unwrap(), 
-        cleanup_thread.unwrap()]
-    )
+    Ok(vec![keep_alive_thread?, agent_job_thread?, cleanup_thread?])
 }
 
 fn main() -> Result<(), Error> {
     // region Init logger
-    let current_exe_patch = env::current_exe().unwrap();
+    let current_exe_patch = env::current_exe()?;
     let parent_path = current_exe_patch.parent().unwrap();
     let log_file = parent_path.join(PREFIX_LOG_NAME);
     let condition = RollingConditionBasic::new().daily();
-    let file_appender = BasicRollingFileAppender::new(log_file, condition, 3).unwrap();
+    let file_appender = BasicRollingFileAppender::new(log_file, condition, 3)?;
     let (file_writer, _guard) = tracing_appender::non_blocking(file_appender);
-    tracing_subscriber::fmt().json().with_writer(file_writer).init();
+    tracing_subscriber::fmt()
+        .json()
+        .with_writer(file_writer)
+        .init();
     // endregion
     // region Process execution
     info!("Starting OpenBAS agent {} ({})", VERSION, Settings::mode());
     let settings = Settings::new();
     let settings_data = settings.unwrap();
     if service_stub::is_windows_service() {
         // Running as a Windows service
-        agent_start(settings_data).unwrap();
+        agent_start(settings_data)?;
```

Comment:
**question**
Are we sure to propage the error instead of unwrap it immediately ?